synapse/handlers/profile.py

```diff
@@ -309,6 +309,16 @@ async def set_avatar_url(
         await self._update_join_states(requester, target_user)
 
     async def on_profile_query(self, args: JsonDict) -> JsonDict:
+        if (
+            not self.hs.config.allow_profile_lookup_over_federation
+            and args.get("origin") != self.hs.hostname
+        ):
```

Comment:
Since this is only called from the federation API I think we can just not bother with the check `args.get("origin") != self.hs.hostname`? I think a) it makes it simpler and b) we are sort of trying to limit the places where we do these sort of checks to make any future vhosting efforts easier.

Since you're there could you also add a docstring to the function, something like:

```python
async def on_profile_query(self, args: JsonDict) -> JsonDict:
    """Handles federation profile query requests.
    """
```